pkg/webhooks/ensurance/validating.go

```diff
@@ -0,0 +1,253 @@
+	if !ok {
+		return fmt.Errorf("req can not convert to NodeQOSEnsurancePolicy")
+	}
+
+	allErrs := genericvalidation.ValidateObjectMeta(&nep.ObjectMeta, false, genericvalidation.NameIsDNS1035Label, field.NewPath("metadata"))
+
+	if nep.Spec.Selector != nil {
+		allErrs = append(allErrs, metavalidation.ValidateLabelSelector(nep.Spec.Selector, field.NewPath("spec").Child("selector"))...)
+	}
+
+	allErrs = append(allErrs, validateNodeQualityProbe(nep.Spec.NodeQualityProbe, field.NewPath("nodeQualityProbe"))...)
+
+	var httpGetEnable bool
+	if nep.Spec.NodeQualityProbe.HTTPGet != nil {
+		httpGetEnable = true
+	}
+	allErrs = append(allErrs, validateObjectiveEnsurances(nep.Spec.ObjectiveEnsurances, field.NewPath("objectiveEnsurances"), httpGetEnable)...)
+
+	if len(allErrs) != 0 {
+		return allErrs.ToAggregate()
+	}
+
+	return nil
+}
+
+func validateNodeQualityProbe(nodeProbe ensuranceapi.NodeQualityProbe, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+
+	if (nodeProbe.NodeLocalGet == nil) && (nodeProbe.HTTPGet == nil) {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("httpGet"), "", "The httpGet and nodeLocalGet must set one"))
+		return allErrors
+	}
+
+	if (nodeProbe.NodeLocalGet != nil) && (nodeProbe.HTTPGet != nil) {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("httpGet"), "", "The httpGet and nodeLocalGet can not set at the same time"))
+		return allErrors
+	}
+
+	if nodeProbe.HTTPGet != nil {
+		allErrors = append(allErrors, validateHTTPGetAction(nodeProbe.HTTPGet, fldPath.Child("httpGet"))...)
+	}
+
+	return allErrors
+}
+
+func validateHTTPGetAction(http *corev1.HTTPGetAction, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	if len(http.Path) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("path"), ""))
+	}
+
+	allErrors = append(allErrors, validatePortNumOrName(http.Port, fldPath.Child("port"))...)
+
+	var supportedHTTPSchemes = sets.NewString(string(core.URISchemeHTTP), string(core.URISchemeHTTPS))
+	if !supportedHTTPSchemes.Has(string(http.Scheme)) {
+		allErrors = append(allErrors, field.NotSupported(fldPath.Child("scheme"), http.Scheme, supportedHTTPSchemes.List()))
+	}
+
+	for _, header := range http.HTTPHeaders {
+		for _, msg := range validation.IsHTTPHeaderName(header.Name) {
+			allErrors = append(allErrors, field.Invalid(fldPath.Child("httpHeaders"), header.Name, msg))
+		}
+	}
+
+	return allErrors
+}
+
+func validateObjectiveEnsurances(objects []ensuranceapi.ObjectiveEnsurance, fldPath *field.Path, httpGetEnable bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(objects) == 0 {
+		return append(allErrs, field.Required(fldPath, ""))
+	}
+
+	allNames := sets.String{}
+	for i, obj := range objects {
+		idxPath := fldPath.Index(i)
+
+		if obj.Name == "" {
+			allErrs = append(allErrs, field.Required(idxPath.Child("name"), ""))
+		} else {
+			for _, msg := range validation.IsDNS1123Label(obj.Name) {
```

Comment:
same with nep name, do we need this to be dns complaint

Reply:
It joint an key for the map, like:

var key = strings.Join([]string{n.Name, v.Name}, ".")

so the nep name and object name all used dns complaint.

Reply:
we should not change user experience, like add constraint to end user because of a implementation challenge.

Reply:
I am fine after checking with devin together, found many k8s object is using same way to validate name.